algorithms/CPlusPlus/Maths/pascal-triangle.cpp

```diff
@@ -0,0 +1,39 @@
+// Example program
+#include <iostream>
+#include <cmath>
+using namespace std;
+
+int main()
+{
+    while(true)
+    {
+        cout << "Demonstrating Pascal's triangle (iterations up to power n)" << endl;
+        char choice;
+        cout << "Do you wish to continue? (Y/y or N/n): ";
+        cin >> choice;
+        if(choice == 'Y' || choice == 'y')
+        {
+            cout << "Enter the desired number till which you want to continue: ";
+            int n;
+            cin >> n;
+            for(int i=0;i<n;i++)
+            {
+                for(int j=0;j<n;j++)
+                {
+                    cout << pow(11,i);

```

Comment:
This is a nice trick to get the nth row of the triangle but will it work for any row for n > 5 ?
 
Given the fact that C++ gives limited size to it's data types and there is no primitive datatype available to hold this huge number ( I don't think even `long long` can hold any such number ), I think it would be a better to either use a larger non-primitive data type ( like arrays ) to hold such numbers in your own representation 
or skip all that hassle and print the numbers without storing them anywhere ( by breaking down the bigger number into series of smaller numbers and printing the series instead ), something like this : -

```
1,  
1, 1 
1, 2, 1 
1, 3, 3, 1 
1, 4, 6, 4, 1 
1, 5, 10, 10, 5, 1 
```